pkg/tlsx/ztls/ztls.go

```diff
@@ -269,6 +268,10 @@ func (c *Client) getConfig(hostname, ip, port string, options clients.ConnectOpt
 
 // tlsHandshakeWithCtx attempts tls handshake with given timeout
 func (c *Client) tlsHandshakeWithTimeout(tlsConn *tls.Conn, ctx context.Context) error {
+	// set read deadline
```

Comment:
the select statement is not used correctly, as it should be used to wait for multiple channels simultaneously. In the current implementation, the tlsConn.Handshake() is called directly within the select statement, which blocks the select until the handshake is completed or an error occurs.